criu/include/restorer.h

```diff
@@ -264,14 +264,21 @@ enum {
 	 * finish it, but by the time it gets woken up the stage
 	 * finished is CR_STATE_RESTORE. The forking stage is
 	 * barrier-ed by the root task, this task is also the one
-	 * that switches the stage (into restoring).
+	 * that switches the stage (into fds).
 	 *
 	 * The above is done to lower the amount of context
 	 * switches from root task to criu and back, since the
 	 * separate forking stage is not needed by criu, it's
 	 * purely to make sure all tasks be in sync.
 	 */
 	CR_STATE_FORKING,
+	/*
+	 * All tasks oppened all their fds. Stage is needed for inotify events
+	 * cleanup. These is also a special stage like CR_STATE_FORKING and is
+	 * barriered by the root task, this task also switches the stage (into
+	 * restoring).
+	 */
+	CR_STATE_FDS,
```

Comment:
I am not sure that we want to add a new stage to fix this minor problem. A new stage will affect performance even if processes don't have inotify descriptors.

Reply:
From second time look, It might be easy to do it without new stage. => will do.

Reply:
upd: Not so easy. For now the closest sync point where all fds for all processes are prepared is after restore_finish_stage(task_entries_local, CR_STATE_RESTORE); in __export_restore_task which is in PIE code. I'm not common with PIE peculiarities, if I use cleanup_current_inotify_events from there I get compile problems:

Error (compel/src/lib/handle-elf-host.c:335): Unexpected undefined symbol: `cleanup_current_inotify_events'. External symbol in PIE?

Some advice will be appreciated.